failed to create a kong client in integration tests against kong-ee 3.0

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When running integration with kong enterprise 3.0 candidate image  `kong/kong-gateway-internal:3.0.x.x-alpine`, KIC failed to create a kong admin api client, with the following error message: 
```
panic: unable to build kong api client: creating workspace: HTTP status 401 (message: "Invalid RBAC credentials")
goroutine 461 [running]:
github.com/kong/kubernetes-ingress-controller/v2/internal/util/test.DeployControllerManagerForCluster.func1()
	/home/runner/work/kubernetes-ingress-controller/kubernetes-ingress-controller/internal/util/test/controller_manager.go:83 +0x29c
created by github.com/kong/kubernetes-ingress-controller/v2/internal/util/test.DeployControllerManagerForCluster
	/home/runner/work/kubernetes-ingress-controller/kubernetes-ingress-controller/internal/util/test/controller_manager.go:79 +0x892
```
https://github.com/Kong/kubernetes-ingress-controller/runs/8174376612?check_suite_focus=true

After some investigation on KIC code, the error happened in creating a workspace in kong, and the kong returned 401 on `POST /workspaces`.

### Expected Behavior

Successfully create a kong admin api client and tests goes on.

### Steps To Reproduce

```markdown
Run "Kong 3.0 compatibility" in Actions tab
```


### Kong Ingress Controller version

latest `main` branch

### Kubernetes version

kind 1.23.2

### Anything else?

When I use manifests in `deploy/single` with Kong 3.0 images, the admin API works. May need some configurations in Kong addon of KTF(https://github.com/Kong/kubernetes-testing-framework) or [Kong/charts](https://github.com/Kong/charts).

--- a/.github/workflows/kong3.0_compatibility.yaml
+++ b/.github/workflows/kong3.0_compatibility.yaml
@@ -75,20 +75,23 @@ jobs:
       id: run_integration_tests
       run: make _test.integration
       continue-on-error: true
+      ## skip enterprise-dbless option.
+      if: ${{ matrix.kong-edition == 'OSS' || matrix.database-mode == 'postgres' }} 
       env:
         TEST_KONG_IMAGE: ${{ env.TEST_KONG_IMAGE }}
         TEST_KONG_TAG: ${{ env.TEST_KONG_TAG }}
         TEST_KONG_ENTERPRISE: ${{ env.TEST_KONG_ENTERPRISE }}
         KONG_LICENSE_DATA: ${{ steps.license.outputs.license }}
-        TEST_DATABASE_MODE: ${{ env.TEST_DATABASE_MODE }}
+        DBMODE: ${{ env.TEST_DATABASE_MODE }}
         TEST_KONG_PULL_USERNAME: ${{ secrets.GHA_DOCKERHUB_PULL_USER }}
         TEST_KONG_PULL_PASSWORD: ${{ secrets.GHA_KONG_ORG_DOCKERHUB_PUBLIC_TOKEN }}
+        NCPU: 1
 
     - name: upload diagnostics
       if: ${{ always() }}
       uses: actions/upload-artifact@v3
       with:
-        name: diagnostics-integration-${{ matrix.database-mode }}-${{ matrix.kong-edtion }}
+        name: diagnostics-integration-${{ matrix.database-mode }}-${{ matrix.kong-edition }}
         path: /tmp/ktf-diag*
         if-no-files-found: ignore
     

